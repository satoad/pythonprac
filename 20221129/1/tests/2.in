class C(metaclass=dump):
    def __init__(self, val):
        self.val = val

    def sub(self, other, another=None):
        return self.val - other - (another or self.val)

c = C(10)
print(c.sub(9))
print(c.sub(9,another=10))
