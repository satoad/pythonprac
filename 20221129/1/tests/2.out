__init__: (10,), {}
sub: (9,), {}
-9
sub: (9,), {'another': 10}
-9
